specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1007 @@
+particular implementation of a Broker. When a Broker is created, the
+`eventing.knative.dev/broker.class` annotation and the `spec.config` field MUST
+be populated (`spec.config` MAY be an empty object) to indicate which of several
+possible Broker implementations to use. It is RECOMMENDED to default the
+`eventing.knative.dev/broker.class` field on creation if it is unpopulated. Once
+created, both fields MUST be immutable; the Broker MUST be deleted and
+re-created to change the implementation class or `spec.config`. This pattern is
+chosen to make it clear that changing the implementation class or `spec.config`
+is not an atomic operation and that any implementation would be likely to result
+in event loss during the transition.
+
+### Trigger Lifecycle
+
+A Trigger MAY be created before the referenced Broker indicated by its
+`spec.broker` field; if the Broker does not currently exist or the Broker's
+`Ready` condition is not `true`, then the Trigger's `Ready` condition MUST be
+`false`, and the reason SHOULD indicate that the corresponding Broker is missing
+or not ready.
+
+The Trigger's controller MUST also set the `status.subscriberUri` field based on
+resolving the `spec.subscriber` field before setting the `Ready` condition to
+`true`. If the `spec.subscriber.ref` field points to a resource which does not
+exist or cannot be resolved via
+[Addressable resolution](#addressable-resolution), the Trigger MUST set the
+`Ready` condition to `false`, and at least one condition MUST indicate the
+reason for the error. The Trigger SHOULD also set `status.subscriberUri` to the
+empty string if the `spec.subscriber.ref` cannot be resolved.
+
+If the Trigger's `spec.delivery.deadLetterSink` field it set, it MUST be
+resolved to a URI and reported in `status.deadLetterSinkUri` in the same manner
+as the `spec.subscriber` field before setting the `Ready` condition to `true`.
+
+Once created, the Trigger's `spec.broker` MUST NOT permit updates; to change the
+`spec.broker`, the Trigger can instead be deleted and re-created. This pattern
+is chosen to make it clear that changing the `spec.broker` is not an atomic
+operation, as it could span multiple storage systems. Changes to
+`spec.subscriber`, `spec.filter` and other fields SHOULD be permitted, as these
+could occur within a single storage system.
+
+When a Trigger becomes associated with a Broker (either due to creating the
+Trigger or the Broker), the Trigger MUST only set the `Ready` condition to
+`true` after the Broker has been configured to send all future events matching
+the `spec.filter` to the Trigger's `spec.subscriber`. The Broker MAY send some
+events to the Trigger's `spec.subscriber` prior to the Trigger's
+`Ready`condition being set to `true`. When a Trigger is deleted, the Broker MAY
+send some additional events to the Trigger's `spec.subscriber` after the
+deletion.
+
+### Channel Lifecycle
+
+A Channel represents an Addressable endpoint (i.e. it MUST have a
```

Comment:
I think MUSTs like this should change so it just says "A Channel MUST be Addressable", and then point to the Addressable section. Then we'll implicitly allow for varying types of Addressable objects... duck-typed ones, K8s Services, or others.